### PR TITLE
TME-2307: add support for existing trail with x-account resources

### PR DESCRIPTION
**Pull request by sarjumulmi**
> Add support for existing cloudtrail with resources deployed in different aws accounts.

Changes:
- `configuration_alias` added for log bucket account. need to pass in the provider alias for log bucket when invoking the module
- conditionally create resources based on `is_existing_cloudtrail_cross_account` var
- local runtime check for assuring other dependent optional variables are provided when `is_existing_cloudtrail_cross_account` is set to true.
- new variable for `expel assume role name` instead of hard coded string. add [type defn](https://github.com/expel-io/terraform-aws-cloudtrail/issues) for `tags`
- change circle ci config as a workaround for `terraform validate` [bug](https://github.com/hashicorp/terraform/issues/28490#issuecomment-1140007955)

### CODE REVIEW

File: examples/existing-cloudtrail/cross-account/terraform.tfvars [hunk 1]
```diff
@@ -0,0 +1,7 @@
+region                                    = "Replace with the AWS region in which you want the notification queue for CloudTrail to be set up"
+expel_customer_organization_guid          = "Replace with your organization GUID assigned to you by Expel. You can find it in your browser URL after navigating to Settings > My Organization in Workbench"
+existing_cloudtrail_bucket_name           = "Replace with your AWS S3 Bucket name"
+aws_mgmt_account_id                       = "Replace with your AWS management account id"
+existing_cloudtrail_log_bucket_account_id = "Replace with your AWS cloudtrail log bucket account id"
```
Comment (bachu20) [inline examples/existing-cloudtrail/cross-account/terraform.tfvars:5]: Nit:
```suggestion
existing_cloudtrail_log_bucket_account_id = "Replace with your AWS CloudTrail log bucket account id"
```

File: variables.tf [hunk 2]
```diff
@@ -12,6 +12,7 @@ variable "prefix" {
 
 variable "tags" {
   description = "A set of tags to group resources."
+  type        = map(string)
```
Comment (bachu20) [inline variables.tf:15]: Sweet, thanks for knocking this out too!

**Review by bachu20 — DISMISSED** [review]
lgtm. Would definitely defer to @davidmontoyago + @ethanmil for reviews before merging it in.

File: README.md [hunk 3]
```diff
@@ -62,13 +63,16 @@ Please contact your Engagement Manager if you have an existing CloudTrail with a
...
-| <a name="input_expel_aws_account_arn"></a> [expel\_aws\_account\_arn](#input\_expel\_aws\_account\_arn) | Expel's AWS Account ARN to allow assuming role to gain CloudTrail access. | `string` | `"arn:aws:iam::012205512454:user/ExpelCloudService"` | no |
+| <a name="input_expel_aws_user_arn"></a> [expel\_aws\_user\_arn](#input\_expel\_aws\_user\_arn) | Expel's AWS User ARN to allow assuming role to gain CloudTrail access. | `string` | `"arn:aws:iam::012205512454:user/ExpelCloudService"` | no |
 | <a name="input_expel_customer_aws_account_id"></a> [expel\_customer\_aws\_account\_id](#input\_expel\_customer\_aws\_account\_id) | Account id of customer's AWS account that will be monitored by Expel if it is different than the one terraform is using. This should be the management account id if organization trail is enabled. | `string` | `null` | no |
+| <a name="input_is_existing_cloudtrail_cross_account"></a> [is\_existing\_cloudtrail\_cross\_account](#input\_is\_existing\_cloudtrail\_cross\_account) | For an existing cloudtrail, whether the cloudtrail & the log bucket (& optinally log bucket notifier topic if existing) are in different aws accounts | `bool` | `false` | no |
```
Comment (sarjumulmi) [inline README.md:70]: this applies to cases where you have an existing organization cloudtrail but the cloudtrail log bucket account is NOT in the management account.

File: variables.tf [hunk 4]
```diff
@@ -50,12 +57,36 @@ variable "enable_organization_trail" {
...
+  default     = false
+}
 variable "existing_cloudtrail_bucket_name" {
   description = "The name of the existing bucket connected to the existing CloudTrail"
   type        = string
   default     = null
 }
 
+variable "aws_mgmt_account_id" {
+  description = "Account id of customer's AWS mgmt account."
```
Comment (ethanmil) [inline variables.tf:73]: mgmt -> management

Comment (ethanmil) [reply variables.tf:73]: omit `customer's`

File: variables.tf [hunk 5]
```diff
@@ -50,12 +57,36 @@ variable "enable_organization_trail" {
...
+  type        = string
+  validation {
+    condition     = var.aws_mgmt_account_id == null || can(regex("^[0-9]{12}$", var.aws_mgmt_account_id))
+    error_message = "Account id must be 12 digits."
+  }
+}
+
+variable "existing_cloudtrail_log_bucket_account_id" {
+  description = "Account id of customer's AWS account where the existing cloudtrail log bucket is created. This is where the new SQS queue will be created"
```
Comment (ethanmil) [inline variables.tf:82]: omit `customer's`

**Review by ethanmil — CHANGES_REQUESTED** [review]
Looking good! Just a few typos and readability suggestions.

**Review by davidmontoyago — COMMENTED** [review]
pair-reviewed with @ethanmil

File: .circleci/config.yml [hunk 6]
```diff
@@ -86,4 +86,17 @@ jobs:
...
+          # As a workaround, we create a temp provider file before running `terraform validate`
+          # https://github.com/hashicorp/terraform/issues/28490#issuecomment-1140007955
+          name: create temp provider file
+          command: |
+            cat > provider.tf \<<EOF
+            provider "aws" {
+              region = "us-east-1"
+              alias  = "log_bucket"
+            }
+            EOF
```
Comment (davidmontoyago) [inline .circleci/config.yml:101]: 💯

**Review by davidmontoyago — APPROVED** [review]
LGTM

**Review by ethanmil — APPROVED** [review]
LGTM

**Review by davidmontoyago — APPROVED** [review]
additions 372aeda and 1a335276dd7b3ed7dfc31616629a44986dc3e22d LGTM

**Pull request merged**